src/main/java/org/folio/oaipmh/service/impl/SetServiceImpl.java

```diff
@@ -82,15 +85,16 @@ public SetServiceImpl(final SetDao setDao) {
   @Override
   public Future<FilteringConditionValueCollection> getFilteringConditions(Map<String, String> okapiHeaders) {
     Promise<FilteringConditionValueCollection> promise = Promise.promise();
-    HttpClient vertxHttpClient = Vertx.vertx()
-      .createHttpClient();
-    Future<JsonObject> locationFuture = getFilteringConditionValues(LOCATION_URI, vertxHttpClient, okapiHeaders);
-    Future<JsonObject> illPoliciesFuture = getFilteringConditionValues(ILL_POLICIES_URI, vertxHttpClient, okapiHeaders);
-    Future<JsonObject> materialTypesFuture = getFilteringConditionValues(MATERIAL_TYPES_URI, vertxHttpClient, okapiHeaders);
-    Future<JsonObject> resourceTypeFuture = getFilteringConditionValues(RESOURCE_TYPES_URI, vertxHttpClient, okapiHeaders);
-    Future<JsonObject> instanceFormatsFuture = getFilteringConditionValues(INSTANCE_FORMATS_URI, vertxHttpClient, okapiHeaders);
-
-    CompositeFuture.all(locationFuture, illPoliciesFuture, materialTypesFuture, resourceTypeFuture, instanceFormatsFuture)
+    WebClient webClient = WebClient.create(Vertx.vertx());
+    Future<JsonObject> locationFuture = getFilteringConditionValues(LOCATION_URI, webClient, okapiHeaders);
+    Future<JsonObject> illPoliciesFuture = getFilteringConditionValues(ILL_POLICIES_URI, webClient, okapiHeaders);
+    Future<JsonObject> materialTypesFuture = getFilteringConditionValues(MATERIAL_TYPES_URI, webClient, okapiHeaders);
+    Future<JsonObject> resourceTypeFuture = getFilteringConditionValues(RESOURCE_TYPES_URI, webClient, okapiHeaders);
+    Future<JsonObject> instanceFormatsFuture = getFilteringConditionValues(INSTANCE_FORMATS_URI, webClient, okapiHeaders);
+
```

Comment:
Another place where WebClient is not closed.

Reply:
Very good point. Thank you, Craig!

Reply:
@craigmcnally, I believe it is closed some lines below. Or I missed something?

  List<Future<JsonObject>> futures = Arrays.asList(locationFuture, illPoliciesFuture, materialTypesFuture,
      resourceTypeFuture, instanceFormatsFuture);
    GenericCompositeFuture.all(futures)
      .onComplete(result -> {
        if (result.failed()) {
          webClient.close();
          promise.fail(result.cause());
        } else {
          List<SetsFilteringCondition> values = new ArrayList<>();
          values.add(jsonObjectToSetsFilteringCondition(locationFuture.result(), LOCATION_JSON_FIELD_PATH, LOCATION));
          values.add(jsonObjectToSetsFilteringCondition(illPoliciesFuture.result(), ILL_POLICIES_JSON_FIELD_PATH, ILL_POLICIES));
          values
            .add(jsonObjectToSetsFilteringCondition(materialTypesFuture.result(), MATERIAL_TYPES_JSON_FIELD_PATH, MATERIAL_TYPES));
          values
            .add(jsonObjectToSetsFilteringCondition(resourceTypeFuture.result(), INSTANCE_TYPES_JSON_FIELD_PATH, INSTANCE_TYPES));
          values.add(jsonObjectToSetsFilteringCondition(instanceFormatsFuture.result(), INSTANCE_FORMATS_JSON_FIELD_PATH,
              INSTANCE_FORMATS));

          FilteringConditionValueCollection filteringConditionValueCollection = new FilteringConditionValueCollection()
            .withSetsFilteringConditions(values);
          webClient.close();
          promise.complete(filteringConditionValueCollection);
        }
      });